SimPEG/Directives.py

```diff
@@ -806,17 +808,16 @@ def initialize(self):
                         "Cannot form the sensitivity explicitely"
                     )
 
-                    m = self.invProb.model
-
-                    JtJdiag += np.sum((dmisfit.W*prob.getJ(m))**2., axis=0)
+                    JtJdiag += np.sum((dmisfit.W * prob.getJ(m))**2., axis=0)
 
                 self.opt.JtJdiag = JtJdiag
 
             # Update the pre-conditioner
             reg_diag = np.zeros_like(self.invProb.model)
             for reg in self.reg.objfcts:
-                reg_diag += self.invProb.beta*(reg.W.T*reg.W).diagonal()
-
+                W = reg.deriv2(m)
+                # reg_diag += self.invProb.beta * (reg.W.T * reg.W).diagonal()
```

Comment:
Thanks @thast, this seems like a more robust way, esp when working with combo objective functions. Can you remove the commented out-line? @fourndo: this is relevant to your stuff, so please take a look when you have a chance!